src/schemas/json/uplugin.json

```diff
@@ -0,0 +1,401 @@
+{
+  "$schema": "http://json-schema.org/draft-04/schema#",
+  "additionalProperties": false,
+  "description": "Descriptor for plugins. Contains all the information contained within a `.uplugin` file.",
+  "definitions": {
+    "BuildConfiguration": {
+      "description": "Available build configurations. Mirorred from `UnrealTargetConfiguration`.",
+      "type": "string",
+      "enum": [
+        "Unknown",
+        "Debug",
+        "DebugGame",
+        "Development",
+        "Shipping",
+        "Test"
+      ]
+    },
+    "BuildTargetType": {
+      "description": "Enumerates build target types.",
+      "type": "string",
+      "enum": ["Unknown", "Game", "Server", "Client", "Editor", "Program"]
+    },
+    "ModuleDescriptor": {
+      "description": "Description of a loadable module.",
+      "type": "object",
+      "properties": {
+        "AdditionalDependencies": {
+          "description": "List of additional dependencies for building this module.",
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        },
+        "HasExplicitPlatforms": {
+          "description": "When true, an empty PlatformAllowList is interpeted as 'no platforms' with the expectation that explict platforms will be added in plugin extensions",
+          "type": "boolean",
+          "default": false
+        },
+        "LoadingPhase": {
+          "description": "When should the module be loaded during the startup sequence? This is sort of an advanced setting.",
+          "type": "string",
+          "enum": [
+            "EarliestPossible",
+            "PostConfigInit",
+            "PostSplashScreen",
+            "PreEarlyLoadingScreen",
+            "PreLoadingScreen",
+            "PreDefault",
+            "Default",
+            "PostDefault",
+            "PostEngineInit",
+            "None",
+            "Max"
+          ]
+        },
+        "Name": {
+          "description": "Name of this module",
+          "type": "string"
+        },
+        "PlatformAllowList": {
+          "description": "List of allowed platforms",
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        },
+        "PlatformDenyList": {
+          "description": "List of disallowed platforms",
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        },
+        "ProgramAllowList": {
+          "description": "List of allowed programs",
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        },
+        "ProgramDenyList": {
+          "description": "List of disallowed programs",
+          "type": "array",
```

Comment:
Optional fix: add `"uniqueItems": true` to make sure that there are no duplicates.

Reply:
Optional fix implemented in 262cb20dcfc106d02304efb65aa070fa0db880cc.